### PR TITLE
Relative path writes for volume get

### PR DESCRIPTION
**Pull request by TheQuantumFractal**
## Details

Very small change to support relative paths for volume / nfs get. Strips the remote path up to the first glob if present then computes file output paths relative to this path.

Closes MOD-2204

<details> <summary>Backward/forward compatibility checks</summary>

---

Check these boxes or delete any item (or this section) if not relevant for this PR.

- [x] Client+Server: this change is compatible with old servers
- [x] Client forward compatibility: this change ensures client can accept data intended for later versions of itself

---

</details>

### CODE REVIEW

File: modal/cli/_download.py [hunk 1]
```diff
@@ -37,7 +37,9 @@ async def producer():
             if is_pipe:
                 await q.put((None, entry))
             else:
-                output_path = local_destination / entry.path
+                start_path = os.path.dirname(remote_path).split("*")[0]
```
Comment (luiscape) [inline modal/cli/_download.py:40]: What is the need for `.split("*")` here? Take the following example:

```python
remote_path = "/my-volume/*.txt
os.path.dirname(remote_path)  # /my-volume
os.path.dirname(remote_path).split("*")[0]  # /my-volume
```

File: modal/cli/_download.py [hunk 2]
```diff
@@ -37,7 +37,9 @@ async def producer():
...
+                rel_path = os.path.relpath(entry.path, start_path.lstrip("/"))
+                output_path = local_destination / rel_path
```
Comment (luiscape) [inline modal/cli/_download.py:42]: Can you use [Path from pathlib](https://docs.python.org/3/library/pathlib.html) instead of `os.path`?

**Review by luiscape — CHANGES_REQUESTED** [review]
Small changes.

File: modal/cli/_download.py (hunk 1 above)
Comment (TheQuantumFractal) [reply modal/cli/_download.py:40]:
```py
remote_path = "/vol1/*/*.txt"
os.path.dirname(remote_path) # /vol1/*
os.path.dirname(remote_path).split("*")[0] # /vol1/
```

os.path.relpath won't work properly if there are still globs in the path (eg if glob for matching folders)

File: modal/cli/_download.py (hunk 2 above)
Comment (TheQuantumFractal) [reply modal/cli/_download.py:42]: Yeah entry.path and start_path are both strings though so I'd have to turn one into a PurePath

**Review by luiscape — APPROVED** [review]

**Pull request merged**